source/components/molecules/RenderCard/RenderCard.tsx

```diff
@@ -0,0 +1,155 @@
+
+/***** types describing how we should send in the data to render our cards  */
+interface Image {
+  type: 'image';
+  image: keyof typeof icons;
+  circle?: boolean;
+  style?: React.CSSProperties;
+}
+
+interface Text {
+  type: 'text';
+  text: string;
+  italic?: boolean;
+}
+
+interface Title {
+  type: 'title';
+  text: string;
+}
+
+interface Subtitle {
+  type: 'subtitle';
+  text: string;
+}
+
+interface Button {
+  type: 'button';
+  text: string;
+  colorSchema?: 'blue' | 'red' | 'green' | 'purple';
+  icon?: string;
+  iconPosition?: 'left' | 'right';
+  action:
+    | {
+        type: 'email';
+        email: string;
+      }
+    | {
+        type: 'phone';
+        phonenumber: string;
+      }
+    | {
+        type: 'navigate';
+        screen: string;
+      }
+    | {
+        type: 'url',
+        url: string;
+    }
+}
+
+type CardComponent = Image | Text | Title | Subtitle | Button;
+/***** end of types */
+
+/** Maps an object to a Card child component */
+const renderCardComponent = (component: CardComponent, navigation: any) => {
+  switch (component.type) {
+    case 'text':
+      return <Card.Text italic={component.italic}>{component.text}</Card.Text>;
+    case 'title':
+      return <Card.Title>{component.text}</Card.Title>;
+    case 'subtitle':
+      return <Card.SubTitle>{component.text}</Card.SubTitle>;
+    case 'image':
+      return (
+        <Card.Image
+          source={icons[component.image]}
+          style={component.style}
+          circle={component.circle}
+        />
+      );
+  }
+
+  // Treat buttons separately, because they have some more complicated behavior
+  if (component.type === 'button') {
+    const { icon, iconPosition, text, action } = component;
+    let onClick: () => void = () => null;
+    switch (action.type) {
+      case 'email':
+        onClick = () => { launchEmail(action.email)};
+        break;
+      case 'phone':
+        onClick = () => { launchPhone(action.phonenumber)};
+        break;
+      case 'navigate':
+        onClick = () => { if (navigation?.navigate) navigation.navigate(action.screen) }; // TODO think about sending parameters here
+        break;
+      case 'url':
+        onClick = () => { Linking.openURL(action.url)};
+        break;
+    }
+    return (
+      <Card.Button onClick={onClick}>
+        {icon && iconPosition && iconPosition === 'left' && <Icon name={icon} />}
+        <TextComponent>{text}</TextComponent>
+        {icon && (!iconPosition || iconPosition === 'right') && <Icon name={icon} />}
+      </Card.Button>
+    );
+  }
+};
+
+interface Props {
+  colorSchema?: 'blue' | 'red' | 'green' | 'purple' | 'neutral';
+  backgroundColor?: 'blue' | 'red' | 'green' | 'purple' | 'neutral';
+  shadow?: boolean;
+  outlined?: boolean;
+  components: CardComponent[];
+}
+
+/** Dynamically renders a card with the sent in children as an array of json objects. */
+const RenderCard: React.FC<Props> = ({
```

Comment:
The naming of this component is quite vague and probably confusing. Since we have a Card component that also renders a Card. I have no suggestion for a better naming, but a suggestion is to find the space for where it will be used and name this component after that.

Reply:
I guess this answers your question about naming :P
Ie i do not have any good suggestion haha.

Reply:
Yeah, I couldn't think of a good name either, it's not so easy. As we discussed, this is almost not a component, but something 'higher order', so the name should reflect that.